src/DefinedSchemas.js

```diff
@@ -0,0 +1,300 @@
+import Parse from 'parse/node';
+import { logger } from './logger';
+import Config from './Config';
+import { internalCreateSchema, internalUpdateSchema } from './Routers/SchemasRouter';
+import { defaultColumns } from './Controllers/SchemaController';
+
+export class DefinedSchemas {
+  constructor(localSchemas, config) {
+    this.config = Config.get(config.appId);
+    this.localSchemas = localSchemas;
+    this.retries = 0;
+    this.maxRetries = 3;
+  }
+
+  // Simulate save like the SDK
+  // We cannot use SDK since routes are disabled
+  async saveSchemaToDB(schema) {
+    const payload = {
+      className: schema.className,
+      fields: schema._fields,
+      indexes: schema._indexes,
+      classLevelPermissions: schema._clp,
+    };
+    await internalCreateSchema(schema.className, payload, this.config);
+    this.resetSchemaOps(schema);
+  }
+
+  async resetSchemaOps(schema) {
+    // Reset ops like SDK
+    schema._fields = {};
+    schema._indexes = {};
+  }
+
+  // Simulate update like the SDK
+  // We cannot use SDK since routes are disabled
+  async updateSchemaToDB(schema) {
+    const payload = {
+      className: schema.className,
+      fields: schema._fields,
+      indexes: schema._indexes,
+      classLevelPermissions: schema._clp,
+    };
+    await internalUpdateSchema(schema.className, payload, this.config);
+    this.resetSchemaOps(schema);
+  }
+
+  async execute() {
```

Comment:
Why can't we just clean up the _Session collection and populate it with the startup data?

Reply:
not sure to understand, what do you mean by `clean up the _Session` ?

Reply:
Sorry. Clean up the _SCHEMA collection.

Reply:
I'm not sure it will work since clean up schema will not perform field deletion, index deletion on the database.

Reply:
I'm not sure if we should delete field/indexes. Deleting a column using the dashboard currently does not perform these operations. Also it could take a long time for a huge class. Why don't we just leave the data over there?

Reply:
I think it's needed here why 🙂 :
- We need to delete index in case of user keep the index name but change the index content
- We need to delete index to avoid unnecessary DB cpu/ram usage
- We need to delete field if field is type is changed
- We need to delete and then recreate field in case of type update 
- We need to delete field when it was removed from the schema to avoid data breach/stale data at DB level, and also free some db storage

> Deleting a column using the dashboard currently does not perform these operations.
Are you sure about this ? Dashboard do not use `deleteField` method ?

Also what happen if user create a field, then 2 days after delete it, and then 3 month after he recreate the field. developer will have some stale data.

I am convinced that we should avoid unnecessary DB maintenance and weird issue on stale data/indexes, we need to keep schema sync with data into the DB 🙂 

>  Also it could take a long time for a huge class. 
May be in another PR we can take a look on how to optimize this part, if needed

Reply:
Got it. I've just checked and it looks that the dashboard is also working like that (differently than what I thought). Another point. I'm afraid that in the case I have multiple parse server instances and restart them all together we can have strange behaviors. Could you add a test case?

Reply:
Yes, this part is managed by the simple retry system. Thanks to the stateless behavior of this new feature, and the poll and diff system which manage fields, indexes, clps in separate way. 

Parallel deploying should not be an issue since each parse server instance will try to perform the schema operations until schema from the db do not fulfill schema provided by the developer (poll/diff advantage) then if an error occur the retry systemr will try to perform again from scratch the schema ops with last changes detected from the DB.

You are right I will add a promise all test to cover this use case correctly

Reply:
I share the concerns regarding auto-deletion of field data and indexes. How would a field of a large collection be deleted without impacting DB performance?

Reply:
Hummm, to be honest currently i don't know the answer and also the solution. I  have checked your linked issue/pr.
I suppose that you have already encountered a performance issue on your projects on a field deletion ?

Deleting a field takes 100% of the database processor and also slows down other requests?
I don't know if mongo/postgres support field delete/unset in background without eating all the CPU or if we should manage this on our side ?

Reply:
The PR describes that deleting a field in Parse Dashboard (as it works today) can be resource intensive for large collections. That is why deleting a field in the DB takes advantage of an index, if there is one, which has to be manually created and removed. The reason the index is not auto-created is because also index creation can be resource intensive for large collections.

Reply:
So here it seems that we are blocked :/

I think in first step, we can add documentation on the issue of running field deletion on large collection. And also raise a warning on field deletion with a link to the docs.

Currently I do not see a solution to ensure data consistency and operation performance for this case :/ 

I'm totally open for an implementation suggestion.

Reply:
Yes, I think a prominent warning in the docs would be helpful. This topics get further complicated when looking at the difference of index building processes between MongoDB <4.4 and >=4.4.

A pragmatic solution could be define a step-by-step process for the developer to manually create additional required indices, then deploy with changed schema, then manually delete unused indices.